src/main/java/uk/nhs/prm/e2etests/service/SqsService.java

```diff
@@ -27,6 +49,86 @@ public SqsService(AwsCredentialsProvider awsCredentialsProvider) {
                 .build();
     }
 
+    /**
+     * This method has a complexity of O(N), where N is
+     * expectedNumberOfEhrCores + expectedNumberOfFragments,
+     * the total number of messages to be received and filtered.
+     * @param queueUri The URI of the Amazon SQS Queue.
+     * @param expectedNumberOfEhrCores The number of cores we are expecting.
+     * @param expectedNumberOfEhrFragments The number of fragments we are expecting.
+     * @param outboundConversationIds The outboundConversationId.
+     * @return If all the messages were found successfully.
+     */
+    public boolean getAllMessagesFromQueue(int expectedNumberOfEhrCores,
+                                           int expectedNumberOfEhrFragments,
+                                           List<String> outboundConversationIds,
+                                           String queueUri) {
+        final CharSequence[] filterCriteria = outboundConversationIds.toArray(new CharSequence[0]);
+        int totalNumberOfMessages = expectedNumberOfEhrCores + expectedNumberOfEhrFragments;
+        final List<Message> allMessages = new ArrayList<>();
+        boolean allMessagesFound = false;
+
+        int emptyResponseCount = 0;
+        int ehrCoreCount = 0;
+        int ehrFragmentCount = 0;
```

Comment:
do you reckon we rename these to `ehrCoreFound `and `ehrFragmentFound ` to be inline with "allMessagesFound" ?